Mathlib/Analysis/Analytic/IsolatedZeros.lean

```diff
@@ -217,6 +217,27 @@ lemma order_eq_nat_iff (hf : AnalyticAt 𝕜 f z₀) (n : ℕ) : hf.order = ↑n
     refine ⟨fun hn ↦ (WithTop.coe_inj.mp hn : h.choose = n) ▸ h.choose_spec, fun h' ↦ ?_⟩
     rw [unique_eventuallyEq_pow_smul_nonzero h.choose_spec h']
 
+/- An analytic function `f` has finite order at a point `z₀` iff locally looks
+  like `(z - z₀) ^ order • g`, where `g` is analytic and does not vanish at
+  `z₀`. -/
+lemma order_neq_top_iff (hf : AnalyticAt 𝕜 f z₀) :
+    hf.order ≠ ⊤ ↔ ∃ (g : 𝕜 → E), AnalyticAt 𝕜 g z₀ ∧ g z₀ ≠ 0
+      ∧ f =ᶠ[𝓝 z₀] fun z ↦ (z - z₀) ^ (hf.order.toNat) • g z := by
+  erw [← hf.order_eq_nat_iff]
+  exact ⟨fun h₁f ↦ (ENat.coe_toNat h₁f).symm, fun h₁f ↦ ENat.coe_toNat_eq_self.mp h₁f.symm ⟩
+
+/- An analytic function has order zero at a point iff it does not vanish there. -/
+lemma order_eq_zero_iff (hf : AnalyticAt 𝕜 f z₀) :
+    hf.order = 0 ↔ f z₀ ≠ 0 := by
+  rw [(by rfl : (0 : ENat) = (0 : Nat)), order_eq_nat_iff hf 0]
+  constructor
+  · intro ⟨g, _, _, hg⟩; simpa [Eventually.self_of_nhds hg]
+  · intro hz; use f; exact ⟨hf, hz, by simp⟩
+
+/- An analytic function vanishes at a point if its order vanishes when converted to ℕ. -/
+lemma zero_if_order_toNat_eq_zero (hf : AnalyticAt 𝕜 f z₀) :
+    hf.order.toNat ≠ 0 → f z₀ = 0 := by simp [hf.order_eq_zero_iff]; tauto
```

Comment:
Purely aesthetic question: does the theorem statement still fit on the same line?
(I'd consider that slightly nicer. This is a matter of taste, i.e. optional for you to follow.)

Reply:
It does. Changed following your suggestion.